Add web section to Concurrency page (#4382)
dart-lang/site-www: src/faq.md

## Native execution

### Q. Is Dart single-threaded?

> No. On native targets, Dart features like isolates and async-await

Reviewer: Async-await does not enable multiple threads. Just mention isolates.

can enable multiple threads of execution at any given time.

> The Dart VM will utilize multiple cores of a multi-core processor
> to run those threads concurrently.

Reviewer 2: **suggestion:** Use basic language and present tense.
```suggestion
The Dart VM uses multiple processor cores
to run those threads concurrently.
```

> [Dart's concurrency architecture](/guides/language/concurrency)
> abstracts the complex, error-prone code of typical shared-memory threading,
> which might explain the misconception that Dart is single-threaded.

Reviewer 2: **suggestion:** This reads like it's burying the lede.

```suggestion
The misconception that Dart runs single-threaded might come from its abstraction of shared-memory threading. [Dart's concurrency architecture](/guides/language/concurrency) simplifies threading.

```

Author: Gonna keep as is, imo the lede is Dart's great concurrency architecture

> This is not quite the case for web apps, though; see
> [Is Dart single-threaded on the web?](#q-is-dart-single-threaded-on-the-web)

Reviewer 2: **suggestion:** Simplify the language here.
```suggestion
Dart differs in how it runs on the Web. To learn more, see
[Is Dart single-threaded on the web?](#q-is-dart-single-threaded-on-the-web)
```

Author: I feel like "Dart differs in how it runs on the web" is another statement that can be taken out of context like the whole "dart is single-threaded" thing that started this.

[No.]({{site.news}}/2015/03/dart-for-entire-web.html)
Dart is designed to compile to JavaScript to run across the modern web.

### Q. Is Dart single-threaded on the web?

Dart is *mostly* single-threaded on the web. 
Web apps can use concurrency features like async-await,

> futures, and streams, but not isolates.

Reviewer: Async-await are not _concurrency_ features.

Maybe just:

> Dart is *mostly* single-threaded on the web, since web apps cannot use isolates.
> Web apps can instead use [web workers] to run code concurrently,
> but web workers have different capabilities and restrictions than isolates,
> and are not as easy or efficient to use. For more information, 
> ...

Reviewer 2: **suggestion:** Simplify again.

> Dart web apps cannot use isolates. To run code concurrently,
> Web apps use [web workers]. 
> Web workers have different capabilities and restrictions than isolates.
> They lack the ease and efficiency of isolates. To learn more, 
> ...

Reviewer: Better!
